### PR TITLE
[Question]: Add e2e test for preselecting the most recently visited dashboard

### LINKED ISSUE #31290: [Safety Net] Improve the flow for adding a question to a dashboard
Epic: https://github.com/metabase/metabase/issues/29666

```[tasklist]
### Tasks
- [ ] https://github.com/metabase/metabase/pull/31797
``` 
- [ ] there doesn’t seem to be an e2e test; we should add them given that the new backend endpoint is used only here
- [ ] permissions edge cases are not e2e-covered, e.g. when you lose access to the most recent dashboard

### PR DESCRIPTION
**Pull request by uladzimirdev**
Closes https://github.com/metabase/metabase/issues/31290

### Description

Covers with test cases
- permissions
- lost access
- happy path

### How to verify

run e2e test locally `question-management`

### Checklist

- [x] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

File: e2e/test/scenarios/question/question-management.cy.spec.js [hunk 1]
```diff
@@ -178,19 +180,100 @@ describe("managing question from the question's details sidebar", () => {
...
+
+                modal().as("modal").findByText("Orders in a dashboard").click();
+
+                cy.get("@modal").should("not.exist");
+                // By default, the dashboard contains one question
+                // After we add a new one, we check there are two questions now
+                cy.get(".DashCard").should("have.length", 2);
+              });
+
+              it("should preselect the most recently visited dashboard", () => {
```
Comment (ranquild) [inline e2e/test/scenarios/question/question-management.cy.spec.js:196]: I don't think it's needed to run this test (and the next one) for different users. Can we just run it for the normal user?

File: e2e/test/scenarios/question/question-management.cy.spec.js [hunk 2]
```diff
@@ -178,19 +180,100 @@ describe("managing question from the question's details sidebar", () => {
...
+                it("should handle lost access", () => {
+                  cy.intercept(
+                    "GET",
+                    "/api/activity/most_recently_viewed_dashboard",
+                  ).as("mostRecentlyViewedDashboard");
+
+                  openQuestionActions();
+                  cy.findByTestId("add-to-dashboard-button").click();
+
+                  cy.wait("@mostRecentlyViewedDashboard").then(
```
Comment (ranquild) [inline e2e/test/scenarios/question/question-management.cy.spec.js:223]: I'd rather not check for exact responses, maybe just for the endpoint being called (cy.wait)

File: e2e/test/scenarios/question/question-management.cy.spec.js [hunk 3]
```diff
@@ -178,19 +180,100 @@ describe("managing question from the question's details sidebar", () => {
...
+                    ({ response }) => {
+                      expect(response.statusCode).to.eq(200);
+                    },
+                  );
 
-              cy.get(".Modal")
-                .as("modal")
-                .findByText("Orders in a dashboard")
-                .click();
+                  findSelectedItem().should(
```
Comment (ranquild) [inline e2e/test/scenarios/question/question-management.cy.spec.js:245]: We should probably assert that it's possible to save the dialog with the pre-selected item if it's not checked in another test

**Review by ranquild — APPROVED** [review]

File: e2e/test/scenarios/question/question-management.cy.spec.js (hunk 2 above)
Comment (uladzimirdev) [reply e2e/test/scenarios/question/question-management.cy.spec.js:223]: I wanted to test also BE responses

- visited - 200
- not visited - 204
- no permissions - 204

File: e2e/test/scenarios/question/question-management.cy.spec.js (hunk 3 above)
Comment (uladzimirdev) [reply e2e/test/scenarios/question/question-management.cy.spec.js:245]: it's is checked in the previous tests, so I didn't want to cover the same several times

File: e2e/test/scenarios/question/question-management.cy.spec.js (hunk 1 above)
Comment (uladzimirdev) [reply e2e/test/scenarios/question/question-management.cy.spec.js:196]: AFAIK test plan assumed we needed to cover permissions, that's why I checked several users. But it totally makes sense to me to test only normal user as permissions are covered by another test now

**Pull request merged**